src/Microsoft.AspNetCore.Mvc/DesignTimeMvcServiceCollectionProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    public class DesignTimeMvcServiceCollectionProvider
+    {
+        public static IServiceCollection PopulateServiceCollection(IServiceCollection services, string assemblyName)
+        {
+            var hostingEnvironment = new HostingEnvironment
+            {
+                ApplicationName = assemblyName
+            };
+
+            services.AddSingleton<IHostingEnvironment>(hostingEnvironment);
+            services.AddMvc();
+
+            return services;
+        }
+
+        private class HostingEnvironment : IHostingEnvironment
```

Comment:
We'd also have to move `HostingEnvironment` to abstractions. https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Hosting/Internal/HostingEnvironment.cs